pkg/controllers/workgenerator/controller.go

```diff
@@ -497,6 +495,8 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 				}
 				oldAppliedStatus := meta.FindStatusCondition(oldWork.Status.Conditions, workapi.ConditionTypeApplied)
 				newAppliedStatus := meta.FindStatusCondition(newWork.Status.Conditions, workapi.ConditionTypeApplied)
+				// we only need to handle the case the applied condition is changed between the new and old work objects otherwise, it
```

Comment:
update the comments as we discussed? we only care if the work is applied or not.